### PR TITLE
Anpassungen der Doku aus Feedback in LV

### LINKED ISSUE #7: Anpassung aus Feedback in LV
Feedback von der LV am 15.05.2024:

**DB:**
- GuestToBookings

**API:**
-OpenAPI / Swagger (.yaml content can be checked with https://editor.swagger.io/)

Er will das so haben:
Filter versteh ich ehrlich gsagt nicht und hab auch nirgends eine Doku gefunden
evtl. meint er das so: https://jsonapi.org/recommendations/#filtering
Noch offen, ob die Filterung auf /rooms passieren soll oder auf /rooms/{id}

- GET /api/v1/rooms 
- GET /api/v1/rooms/{id}
- GET /api/v1/rooms/{id}?filter=[{field:startdate, enddate}] <- 404/200
- POST api/v1/bookings <- 201
- PUT api/v1/bookings/{id} <- 200
- DELETE api/v1/bookings/{id} <-  200

**Paperprototype:**
- Währung weg
- Galerie fraglich
- Pagination bei "Zimmerliste"
- Buchungsvorgang (Mockup)

### PR DESCRIPTION
**Pull request by StroSte**
closes #7

### CODE REVIEW

**Comment by StroSte** [comment]
Noch offene Punkte:
- [x] Entscheidung: Filterung auf /rooms passieren soll oder auf /rooms/{id}
- [ ] PUT api/v1/bookings/{id} <- 200
- [ ] DELETE api/v1/bookings/{id} <- 200
- [x] Paperprototype<: noch alles (Kein Zugang)

File: api_specification.yaml [hunk 1]
```diff
@@ -0,0 +1,232 @@
+openapi: 3.0.3
+info:
+  title: Hotel Booking API
+  version: 1.0.0
+  description: API for booking hotel rooms and checking availability.
+paths:
+  /rooms:
```
Comment (lukasfoidl) [inline api_specification.yaml:9]: /api/v1/rooms or add in info the standard path

File: api_specification.yaml [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+
+  /rooms/{id}:
```
Comment (lukasfoidl) [inline api_specification.yaml:79]: /api/v1/rooms/{id} or add in info the standard path

File: api_specification.yaml [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+
+  /booking:
```
Comment (lukasfoidl) [inline api_specification.yaml]: /api/v1/bookings (plural) or add in info the standard path

**Review by lukasfoidl — CHANGES_REQUESTED** [review]
I would add the filter option to both, all rooms and one specific room to request all rooms available in a specific time frame and check one specific room for availability if necessary. If our supervirsor wishes the use of the filter keyword please use it. 

I don't see any reason in the user stories to have a put and delete endpoint on bookings, why is it in the ticket description actually?

Please add the paper prototype changes as well.

**Comment by StroSte** [comment]
@se23m044 just added the updated paperprototype

File: api_specification.yaml [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+                    type: integer
+                    example: 10
+                  offset:
+                    type: integer
+                    example: 5
+                  rooms:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Room'
+        401:
```
Comment (lukasfoidl) [inline api_specification.yaml:64]: Please add 404 error if no rooms found in the specified date range.

**Review by lukasfoidl — CHANGES_REQUESTED** [review]

**Review by lukasfoidl — APPROVED** [review]

**Pull request merged**